src/app.js

```diff
@@ -0,0 +1,247 @@
+                    }
+                }
+            }
+        }
+    },
+    events: {
+        onAppReady: null
+    },
+    listeners: {
+        "onCreate.updateTaskTray": {
+            funcName: "{that}.updateTaskTray",
+            args: ["{that}.model.menu"]
+        }
+    },
+    modelListeners: {
+        "menu": {
+            funcName: "{that}.updateTaskTray",
+            args: ["{change}.value"],
+            excludeSource: "init"
+        }
+    },
+    invokers: {
+        updateTaskTray: {
+            funcName: "gpii.taskTray.updateTaskTray",
+            args: ["{that}.tray", "{that}.options.labels.tooltip", "{arguments}.0"] // menu
+        }
+    },
+    icon: "icons/gpii.ico",
+    labels: {
+        tooltip: "GPII Electron"
+    }
+});
+
+gpii.taskTray.makeTray = function (icon) {
+    return new Tray(path.join(__dirname, icon));
+};
+
+gpii.taskTray.updateTaskTray = function (tray, tooltipLabel, menu) {
+    tray.setToolTip(tooltipLabel);
+    tray.setContextMenu(Menu.buildFromTemplate(menu));
+};
+
+/*
+ ** Component to start and stop parts of the GPII as well as key in and key out users
+ */
+fluid.defaults("gpii.app", {
+    gradeNames: "fluid.modelComponent",
+    model: {
+        gpiiStarted: false,
+        keyedInSet: null
+    },
+    events: {
+        onAppReady: null
+    },
+    listeners: {
+        "onCreate.startGpii": {
+            funcName: "gpii.app.startLocalFlowManager",
+            args: ["{that}"]
+        },
+        "onCreate.changeStarted": {
+            funcName: "{that}.changeStarted",
+            args: [true]
+        }
+    },
+    modelListeners: {
+        keyedInSet: [{
+            funcName: "gpii.app.keyOut",
+            args: ["{change}.oldValue"]
+        }, {
+            funcName: "gpii.app.keyIn",
+            args: ["{change}.value"],
+            priority: "after:keyOut"
+        }]
+    },
+    invokers: {
+        changeStarted: {
+            changePath: "gpiiStarted",
+            value: "{arguments}.0"
+        }
+    }
+});
+
+gpii.app.startLocalFlowManager = function (that) {
+    var fluid = require("universal"),
+        gpii = fluid.registerNamespace("gpii");
+
+    if (os.platform() === "win32") {
+        require("gpii-windows/index.js");
+    }
+
+    gpii.start();
+    //TODO: needs to fire an on ready so we can be certain that everything has started
+    that.events.onAppReady.fire();
+};
+
+gpii.app.keyIn = function (token) {
+    request("http://localhost:8081/user/" + token + "/login", function (/*error, response, body*/) {
+        //TODO Put in some error logging
+    });
+};
+
+gpii.app.keyOut = function (token) {
+    request("http://localhost:8081/user/" + token + "/logout", function (/*error, response, body*/) {
+        //TODO Put in some error logging
+    });
+};
+
+/*
+ ** Component to create and update the task tray menu.
+ */
+fluid.defaults("gpii.taskTray.menu", {
+    gradeNames: "fluid.modelComponent",
+    menuLabels: {
```

Comment:
Try to declaratively encode the effects of the menu items as well as just the labels. It will give your users a central point of control and clean up your code considerably.